hazelcast/src/main/java/com/hazelcast/map/ExtendedValue.java

```diff
@@ -19,12 +19,11 @@
 import static com.hazelcast.util.Preconditions.isNotNull;
 
 /**
- * Represents a value with an expiration time attached to it.
- * Expiration time is optional. It is intented to use with
- * {@link EntryLoader}.
+ * Represents a value with an expiration time attached to it. Expiration
+ * time is optional. It is intended to be used with {@link EntryLoader}.
  * @param <V> the type of the value object
  */
-public class EntryLoaderEntry<V> {
+public class ExtendedValue<V> {
```

Comment:
The name `ExtendedValue` feels a bit too generic to me. Without further context, it it unclear what its purpose is, and combined with the fact that it sits directly in `com.hazelcast.map`, it may look more important that it is. The Javadoc clarifies that but still... Since this class it tightly bound to `MapLoader`/`MapStore`, what about making it a nested class of the `MapLoader` interface, or at least change its name to something like MapLoaderValue`? (Having said that, I know you played with different ideas in older commits.)

Reply:
Movement to map package is handled here: https://github.com/hazelcast/hazelcast/pull/15149

Reply:
alternative name suggestion: 
since we use ExtendedValue in both store and load mechanism, `MetaDataAwareValue` can be another alternative.

Reply:
I appreciate the ideas about naming. I was actually going to go with `MapLoaderValue` but I did not because of the reason Ahmet stated. I think `MetadataAware` value sounds good. Plus, we might include more metadata in it in the future if requested. I will also include it as a nested class inside NOT `MapLoader` but `EntryLoader` as I think @vojtechtoman meant to suggest.